lms/templates/dashboard.html

```diff
@@ -183,7 +183,7 @@
                     pseudo_session = unfulfilled_entitlement_pseudo_sessions[str(entitlement.uuid)]
                     if not pseudo_session:
                         continue
-                    enrollment = CourseEnrollment(user=user, course=CourseOverview.get_from_id(pseudo_session['key']), mode=pseudo_session['type'])
+                    enrollment = CourseEnrollment(user=user, course=CourseOverview.get_from_id(CourseKey.from_string(pseudo_session['key'])), mode=pseudo_session['type'])
```

Comment:
You'll want to update this one as well: `themes/edx.org/lms/templates/dashboard.html`

Also, when I made this change to wrap the `pseudo_session['key']` in a `CourseOverview.get_from_id` most of the errors we were getting in production showed that the pseude_session['key'] was an opaque key.  So I think you'll want to do a check to see if is or isn't a coureskey before you try to make this conversion.

Reply:
I am trying to recreate this locally right now to test it more.

Reply:
Everywhere I can see in the code that sets psuedo_session['key'] is a dict of the course run data from discovery. Also elsewhere in this file whenever we retrieve the id in this way we create the opaque key.  I will add a check for now just to be safe.  And then add some logging to come back later and review.

So far in local testing things are working without these changes, but this is complicated code.